README.md

```diff
@@ -166,6 +166,26 @@ For more information, see:
 [Metrics API design]: https://github.com/kubernetes/design-proposals-archive/blob/main/instrumentation/resource-metrics-api.md
 [Metrics Server design]: https://github.com/kubernetes/design-proposals-archive/blob/main/instrumentation/metrics-server.md
 
+This diagram shows how metrics-server handles a `kubectl top pods` request:
+```mermaid
+sequenceDiagram
+    participant Client
+    participant APIServer
+    participant MS as api.podMetrics
+    participant Stg as storage.Storage
+    participant PL as cache.GenericLister (Pods)
+
+    Client->>APIServer: GET /apis/metrics.k8s.io/v1beta1/namespaces/{ns}/pods[/{podName}]
+    APIServer->>MS: List(ctx, opts) or Get(ctx, name, opts)
```

Comment:
Could we simplify this diagram to just show List and skip mention of Get. At the top you mention `kubectl top pods` which will do `List` as pod name is not specified.

Reply:
simplified